Models/ProjectModels/Admin/AdminModel.php

```diff
@@ -0,0 +1,218 @@
+            $this->msgModel->setMsg('empty', 'no_admins', 'empty');
+
+            return null;
+        }
+
+        return $dbResult;
+    }
+
+    /**
+     * @param string $actionName
+     * @return string[]
+     */
+    protected function getConditionData(string $actionName): array
+    {
+        if (!array_key_exists($actionName, self::CONDITION_DATA)) {
+            throw new \InvalidArgumentException('You forgot to add field with value in const CONDITION_DATA!');
+        }
+
+        return self::CONDITION_DATA[$actionName];
+    }
+
+    /**
+     * @param string $actionName
+     * @return string[]
+     */
+    protected function getDbFields(string $actionName): array
+    {
+        if (!array_key_exists($actionName, self::DB_FIELDS)) {
+            throw new \InvalidArgumentException('You forgot to add field with value in const DB_FIELDS!');
+        }
+
+        return self::DB_FIELDS[$actionName];
+    }
+
+    /**
+     * @param array $data
+     * @param string $fieldName
+     * @throws \Exception
+     */
+    public function changeAccess(array $data, string $fieldName): void
+    {
+        $count = count($data);
+        if ($fieldName === 'is_head' && $count > 1) {
+            $this->msgModel->setMsg('failure', 'too_much_admins', 'too_much_admins');
+
+            return;
+        }
+        
+        $conditionData = $this->formatArrayData(['id'], $data);
+        $this->setDbMsgModel();
+        $dbResult = $this->db->select(['id'])->from(['admins'])->condition($conditionData)->query()->fetchAll();
+        if (!$dbResult) {
+            throw new \Exception(
+                'Wrong Admin id:' . $data['id'] . ', during changeAccess->selectAdmins methods!'
+            );
+        }
+
+        $updateData = $this->formatArrayData([$fieldName], $data);
+        $dbResult = $this->db->update(['admins'], $updateData)->condition($conditionData)->exec();
+        if ($dbResult) {
+            $this->msgModel->setMsg('success', $this->getRefererAction(), 'success');
+            if ($fieldName === 'is_head') {
+                $conditionData = $this->formatArrayData(['id'], $this->sessionModel->getUserData());
+                $dbResult = $this->db->update(['admins'], ['is_head' => 0])->condition($conditionData)->exec();
+                if ($dbResult) {
+                    $this->sessionModel->setUserData(['is_head' => '0']);
+                } else {
+                    $this->logger->log(
+                        'default',
+                        'Problems to update db data of previous HeadAdmin '.
+                        $this->sessionModel->getUserData()['name'] . ':(' .
+                        $this->sessionModel->getUserData()['login'] . ')!'
+                    );
+                }
+            }
+        } else {
+            throw new \Exception(
+                'Something went wrong during updating Admin DB data,
+                 using : ' . $this->getRequestAction() . 'Action->changeAccess methods!'
```

Comment:
please use https://www.php.net/manual/en/exception.gettraceasstring.php
and write down trance in logger itself if needed

Reply:
I forgot that I already have getTraceAsString() in  Models\ProjectModels\Exception\Manager->createExceptionMessage();
I will remove all unnecessary in from exception messages all over the project.